### PR TITLE
Print stacktraces to files, instead of stdout

### PR DESCRIPTION
**Pull request by tpasternak**

### CODE REVIEW

**Comment by alexarchambault** [comment]
I pushed a few commits to:
- not mess with today's `ScalaCli.scala` much in the diff
- try to use a sensible workspace value when possible (only with the `run` command, the other commands need to be updated too)
- keep the exception causes in the log (quite important, many times, the relevant exception is in a cause)
- take into account verbosity (keep throwing in the current console if verbosity >= 2, only with the `run` command too, the other commands need to be updated there too)

Feel to squash them / push force on this branch.

**Review by alexarchambault — COMMENTED** [review]
I pushed a few commits to:
- not mess with today's `ScalaCli.scala` much in the diff
- try to use a sensible workspace value when possible (only with the `run` command, the other commands need to be updated too)
- keep the exception causes in the log (quite important, many times, the relevant exception is in a cause)
- take into account verbosity (keep throwing in the current console if verbosity >= 2, only with the `run` command too, the other commands need to be updated there too)

Feel to squash them / push force on this branch.

**Comment by tpasternak** [comment]
@alexarchambault so should I squash and merge it as is?

**Comment by alexarchambault** [comment]
> so should I squash and merge it as is?

Just pushed one more commit. ~I guess it can be squashed-and-merged, yes.~

**Comment by alexarchambault** [comment]
We need to report the changes in `Run.scala` (setting things in `CurrentParams`) to the other commands, prior to merging this.

**Comment by tpasternak** [comment]
I had to rebase due to conflicts

**Comment by alexarchambault** [comment]
@tpasternak Just pushed a commit to factor `-v` out of `LoggingOptions`, so that it can be added to commands without adding all the other logging options (which would be unused).

Good to merge for me, if that's fine for you (and the CI is green).

**Review by alexarchambault — APPROVED** [review]

**Comment by alexarchambault** [comment]
@tpasternak It seems you discarded one of the commits I had pushed, factoring `-v` / `--verbose`, so that we don't add all the logging options when they're not needed.

I just pushed my version of this branch [here](https://github.com/alexarchambault/scala-cli/tree/hide-non-recoverable-exceptions), its last commit is missing in particular (and makes sense for us here IMO).

**Comment by tpasternak** [comment]
Right, I rebased and squashed, as the history became quite complex

File: modules/cli/src/main/scala/scala/cli/commands/InstallHomeOptions.scala [hunk 1]
```diff
@@ -5,6 +5,8 @@ import caseapp._
 // format: off
 @HelpMessage("Install `scala-cli` in a sub-directory of the home directory")
 final case class InstallHomeOptions(
+  @Recurse
+    shared: SharedOptions = SharedOptions(),
```
Comment (alexarchambault) [inline modules/cli/src/main/scala/scala/cli/commands/InstallHomeOptions.scala]: This one should be changed to `VerbosityOptions` I think.

File: modules/cli/src/main/scala/scala/cli/commands/Metabrowse.scala [hunk 2]
```diff
@@ -19,8 +20,9 @@ object Metabrowse extends ScalaCommand[MetabrowseOptions] {
   override def sharedOptions(options: MetabrowseOptions) = Some(options.shared)
 
   def run(options: MetabrowseOptions, args: RemainingArgs): Unit = {
-
+    CurrentParams.verbosity = options.shared.logging.verbosity
     val inputs = options.shared.inputsOrExit(args)
+    CurrentParams.verbosity = options.shared.logging.verbosity
```
Comment (alexarchambault) [inline modules/cli/src/main/scala/scala/cli/commands/Metabrowse.scala]: Seems we want to set the workspace on that line.

**Review by alexarchambault — APPROVED** [review]

**Pull request merged**